herd/AArch64Sem.ml

```diff
@@ -1333,6 +1325,28 @@ module Make
         let ops = List.map op (Misc.interval 0 (neon_nelem (List.hd rlist))) in
         List.fold_right (>>::) ops (M.unitT [[()]])
 
+      (* Data cache operations *)
+      let dc_loc op a ii =
+        let mk_act loc = Act.DC (op,Some loc) in
+        let loc = A.Location_global a in
+        M.mk_singleton_es (mk_act loc) ii
+
+      let do_dc op rd ii =
+        if AArch64Base.DC.sw op then
+          M.mk_singleton_es (Act.DC (op, None)) ii >>! B.Next
+        else begin
+            (* TODO: The size for DC should be a cache line *)
```

Comment:
Perhaps we can rely on the assumption: different variables are allocated to different cache lines? If you need t allocated variables to the same cache line, we can use arrays and then decide whether two cells are in the same cache line or not, assuming  alignment of arrays (_i.e._ alignment according to total array size).

Reply:
I think so. I am not very familiar with the specifics of mixed mode. 

In non-mixed mode: All accesses to a memory location have to be of a specific size, don't they (e.g, byte, half word, ...)? In the mixed mode then this assumption goes away. If that's the case, we can make the same assumption for DC instructions; in non-mixed mode, when we clean location `x` we use the size of all the previous or next memory instruction that accesses `x`.

In mixed mode: We have to be able to handle the case where `x` and `y` (or x[0] and x[1]) map into the same cache line, I think we have most of the mechanics in place, maybe we also need to be able to reason about allocation alignment.

Reply:
I do not know yet. Anyway I'll merge this PR.